File: src/main/java/fi/metatavu/metaform/server/persistence/dao/TableReplyFieldDAO.java
package fi.metatavu.metaform.server.persistence.dao;

import java.util.UUID;

import javax.enterprise.context.ApplicationScoped;

import fi.metatavu.metaform.server.persistence.model.TableReplyField;
import fi.metatavu.metaform.server.persistence.model.Reply;

/**
 * DAO class for TableReplyField entity
 * 
 * @author Antti Leppä
 */
@ApplicationScoped
public class TableReplyFieldDAO extends ReplyFieldDAO<TableReplyField> {
  
  /**
   * Creates new TableReplyField 
   * 
   * @param id id
   * @param reply reply
   * @param name field name
   * @param value field value
   * @return created field
   */
  public TableReplyField create(UUID id, Reply reply, String name) {
    TableReplyField replyField = new TableReplyField(); 
    replyField.setId(id);
    replyField.setName(name);
    replyField.setReply(reply);
    return persist(replyField);
  }
  
}

File: src/main/kotlin/fi/metatavu/metaform/server/persistence/dao/TableReplyFieldDAO.kt
package fi.metatavu.metaform.server.persistence.dao

import fi.metatavu.metaform.server.persistence.model.Reply
import fi.metatavu.metaform.server.persistence.model.TableReplyField
import java.util.*
import javax.enterprise.context.ApplicationScoped

/**
 * DAO class for TableReplyField entity
 *
 * @author Antti Leppä
 */
@ApplicationScoped
class TableReplyFieldDAO : ReplyFieldDAO<TableReplyField>() {
  /**
   * Creates new TableReplyField
   *
   * @param id id
   * @param reply reply
   * @param name field name
   * @param value field value
   * @return created field
   */
  fun create(id: UUID?, reply: Reply, name: String): TableReplyField {
    val replyField = TableReplyField()
    replyField.id = id
    replyField.name = name
    replyField.reply = reply
    return persist(replyField)
  }
}